cli.py

```diff
@@ -123,19 +123,29 @@ def setup():
                 headers = {"Authorization": f"token {github_access_token}"}
                 response = requests.get("https://api.github.com/user", headers=headers)
                 if response.status_code == 200:
-                    scopes = response.headers.get("X-OAuth-Scopes")
+                    o_auth_scopes_header = "X-OAuth-Scopes"
+                    scopes = response.headers.get(o_auth_scopes_header, [])
                     if "public_repo" in scopes or "repo" in scopes:
```

Comment:
**Suggestion:** The check for the presence of the `o_auth_scopes_header` in the response headers uses a list as a default fallback value when calling `get` on the headers dictionary. This could lead to unexpected behavior since the `in` operator checks for the presence of a string within a list, which is not the intended use case. It should be corrected to use a string split by commas to properly handle the scope strings. [bug]
```suggestion
                    o_auth_scopes_header = "X-OAuth-Scopes"
                    scopes = response.headers.get(o_auth_scopes_header, "")
                    if "public_repo" in scopes.split(",") or "repo" in scopes.split(","):
```